Update 01565_reconnect_after_client_error to not expect explicit reconnect

azat opened the pull request:
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

There is no need in explicit reconnect in case of client errors
(#19353), so just rewrite the test to ensure that everything works.

Cc: @kssenii

azat committed:
Fix 01565_reconnect_after_client_error test (wrong expect code for multi-line mode)

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
Fix timeouts in 01565_reconnect_after_client_error

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

kssenii commented on src/Client/ClientBase.cpp:
```diff
@@ -2278,8 +2278,7 @@ void ClientBase::runInteractive()
             /// Client-side exception during query execution can result in the loss of
             /// sync in the connection protocol.
             /// So we reconnect and allow to enter the next query.
-            if (!connection->checkConnected(connection_parameters.timeouts))
```
but I'd assume in this case we would have ping fail here:
https://github.com/ClickHouse/ClickHouse/blob/4d23f74a5347f8ed2126c999e2bcd2ac5a08760e/src/Client/Connection.h#L129
and reconnect anyway. So it is incorrect?

azat replied:
The connection had not been broken in this case, the problem was with the client, from #19353

```
INSERT INTO t0(c0, c1) VALUES ("1",1) ;
-- Code: 47, e.displayText() = DB::Exception: Missing columns: '1' while processing query: '`1`', required columns: '1' '1': data for INSERT was parsed from query, Stack trace (when copying this message, always include the lines below):

INSERT INTO t0(c0, c1) VALUES (1,1) ;
-- Error on processing query: INSERT INTO t0(c0, c1) VALUES ("1",1) ;
-- Code: 47, e.displayText() = DB::Exception: Missing columns: '1' while processing query: '`1`', required columns: '1' '1': data for INSERT was parsed from query, Stack trace (when copying this message, always include the lines below):
```

So it was incorrectly processing the query with the error again, it had been fixed already, so let's just update the test.

azat committed:
Replace 01565_reconnect_after_client_error with 01565_query_loop_after_client_error

There is no need in explicit reconnect in case of client errors
(#19353), so just rewrite the test to ensure that everything works.

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>
